From 07e2e4ddcdfb96bed232db2bc330e578cd05225f Mon Sep 17 00:00:00 2001
From: Andrey Butirsky <butirsky@gmail.com>
Date: Wed, 21 Sep 2022 18:34:04 +0300
Subject: [PATCH] Set QSurfaceFormat::ResetNotification

This got lost in frameworks porting from shared KDeclarative code and is
important especially for nvidia after suspend+resume.
---
 greeter/main.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/greeter/main.cpp b/greeter/main.cpp
index d9efbe1..f252097 100644
--- a/greeter/main.cpp
+++ b/greeter/main.cpp
@@ -21,6 +21,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <QCommandLineParser>
 #include <QDateTime>
 #include <QSessionManager>
+#include <QSurfaceFormat>
 
 #include <iostream>
 
@@ -94,6 +95,10 @@ int main(int argc, char *argv[])
     // Suppresses modal warnings about unwritable configuration files which may render the system inaccessible
     qputenv("KDE_HOME_READONLY", "1");
 
+    auto format = QSurfaceFormat::defaultFormat();
+    format.setOption(QSurfaceFormat::ResetNotification);
+    QSurfaceFormat::setDefaultFormat(format);
+
     ScreenLocker::UnlockApp app(argc, argv);
     app.setQuitOnLastWindowClosed(false);
     QCoreApplication::setApplicationName(QStringLiteral("kscreenlocker_greet"));
-- 
GitLab

